src/model/calculations.ts

```diff
@@ -100,11 +101,154 @@ export class OptionsRemainingCalculator {
   }
 }
 
+interface StockClass {
+  object_type: string;
+  id: string;
+  name: string;
+  class_type: string;
+  default_id_prefix?: string;
+  current_shares_authorized?: string;
+  board_approval_date?: string;
+  votes_per_share?: number;
+  conversion_rights?: StockClassConversionRight[];
+}
+
+interface StockClassConversionRight {
+  conversion_mechanism: {
+    type: string;
+    ratio: {
+      numerator: string;
+      denominator: string;
+    };
+  };
+  rounding_type?: string;
+  converts_to_stock_class_id?: string;
+}
+
+export class ConversionRatioCalculator {
+  private stockClasses: StockClass[] = [];
+
+  public findRatio(preferredStockClassId: string): {
+    ratio: number;
+    lowestVotesPerShare: number | undefined;
+    path: StockClass[];
+  } {
+    const visited: Set<string> = new Set();
+    let lowestVotesPerShare: number | undefined;
```

Comment:
**Non-blocking:** You can simplify your code here by eliminating the need for handling `undefined` by initializing as:

`let lowestVotesPerShare: number = Infinity`